types/core.tst.ts

```diff
@@ -0,0 +1,260 @@
+import { expect, test, describe } from 'tstyche'
+import {
+  encodeCursor,
+  decodeCursor,
+  PAGINATION_MODE,
+  connectionDirective
+} from './core.js'
+import type { ConnectionArgs, ConnectionResolverResponse } from './core.js'
+
+describe('encodeCursor function', () => {
+  test('should return string for string id', () => {
+    expect(encodeCursor('User', '123')).type.toBe<string>()
+  })
+
+  test('should return string for number id', () => {
+    expect(encodeCursor('Post', 456)).type.toBe<string>()
+  })
+})
+
+describe('decodeCursor function', () => {
+  test('should return object with typeName and id', () => {
+    expect(decodeCursor('abc123')).type.toBe<{ typeName: string; id: string }>()
+  })
+})
+
+describe('ConnectionArgs type', () => {
+  test('should accept first and after properties', () => {
+    expect({
+      first: 10,
+      after: 'cursor123'
+    }).type.toBeAssignableTo<ConnectionArgs>()
+  })
+
+  test('should accept only first property', () => {
+    expect({ first: 5 }).type.toBeAssignableTo<ConnectionArgs>()
```

Comment:
Here an in similar cases better use `.toBeAssignableWith()`:

```suggestion
    expect<ConnectionArgs>().type.toBeAssignableWith({ first: 5 })
```

This way the value under test will always stay on the left hand side. Easier to read. Otherwise it does the same job.